### PR TITLE
[FEATURE] Trust project option

### PR DESCRIPTION
**Pull request by pblottiere**
## Description

This PR adds the **trustProject** option.

When this option is activated:
- extent for vector layers is directly read from the project file when the data provider has no metadata (for example in case of PostgreSQL views or materialized views)
- primary key unicity is not checked in the PostgreSQL data provider

This option is convenient for the server in order to avoid spending to much time in **getCapabilities** when we know that the underlying data do not change.

Some tests has been added to confirm the behavior on PostgreSQL views and materialized views.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by haubourg** [comment]
Note: this option will grealty speed up project loading either on Desktop or Server Context for big datasets accessed via views, materialized views or any data format not storing such metadata.  Downsides are that if datasource has effectively changed, Project will need to be updated

**Comment by nyalldawson** [comment]
Shouldn't this be set in the project properties rather than as a global option?

**Comment by pblottiere** [comment]
> Shouldn't this be set in the project properties rather than as a global option?

Good point... @haubourg do you agree?

**Comment by haubourg** [comment]
+1 !

File: src/core/qgsproject.h [hunk 1]
```diff
@@ -755,6 +755,32 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
...
+     * read from the XML document when data source has no metadata or if the
+     * data provider has to determine it. Moreover, when this option is
+     * activated, primary key unicity is not checked for views and
+     * materialized views with Postgres provider.
+     *
+     * \param trust True to trust the project, false otherwise
+     *
+     * \since QGIS 3.0
+     */
+    void setTrust( bool trust );
```
Comment (nyalldawson) [inline src/core/qgsproject.h]: setTrustLayerMetadata/trustLayerMetadata would be more descriptive names here

File: src/ui/qgsoptionsbase.ui [hunk 2]
```diff
@@ -2005,7 +1951,7 @@
                   <item>
                    <widget class="QgsCollapsibleGroupBox" name="mSimplifyDrawingGroupBox">
                     <property name="title">
-                     <string>Enable feature si&amp;mplification by default for newly added layers</string>
+                     <string>Enable fea&amp;ture simplification by default for newly added layers</string>
```
Comment (nyalldawson) [inline src/ui/qgsoptionsbase.ui]: Are these shortcut changes deliberate?

Comment (pblottiere) [reply src/ui/qgsoptionsbase.ui]: Absolutely not, my qt-designer made something strange here...

Comment (pblottiere) [reply src/ui/qgsoptionsbase.ui]: As soon as I open/save this file with my Qt-Designer, shortcuts are changed... Anyway, as *trustProject* is not a global option anymore, I reverted changes on *qgsoptionsbase.ui *.

**Comment by pblottiere** [comment]
I'll merge this PR tomorrow if I've no more comments in the meantime.

**Comment by haubourg** [comment]
In project properties, the option would maybe be better placed in the "Data Sources" tab. 
BTW, renaming "Data Sources" to "Data sources" would be nice :)

**Comment by haubourg** [comment]
Beyond that, I couldn't find any problem in editing / updating extent workflow. 
Editing from QGIS updates it automatically which is good. (Maybe with 1 trillion rows it would be nice to skip that too, let's see when his happens). 
Changing data from the DB, QGIS with trust option will not change the extent unless pressing the "update extent button". 
So +1 for me!

**Comment by pblottiere** [comment]
> In project properties, the option would may be better placed in the "Data Sources" tab.

Done

**Pull request merged**